src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/oracle/PMTransactionDataProvider.kt

```diff
@@ -0,0 +1,112 @@
+package it.pagopa.ecommerce.helpdesk.dataproviders.oracle
+
+import io.r2dbc.spi.ConnectionFactory
+import it.pagopa.ecommerce.helpdesk.dataproviders.TransactionDataProvider
+import it.pagopa.ecommerce.helpdesk.exceptions.InvalidSearchCriteriaException
+import it.pagopa.ecommerce.helpdesk.exceptions.NoResultFoundException
+import it.pagopa.ecommerce.helpdesk.utils.resultToTransactionInfoDto
+import it.pagopa.generated.ecommerce.helpdesk.model.*
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.kotlin.core.publisher.toMono
+
+/**
+ * TransactionDataProvider implementation that search transactions into PM DB
+ *
+ * @see TransactionDataProvider
+ */
+@Component
+class PMTransactionDataProvider(@Autowired private val connectionFactory: ConnectionFactory) :
+    TransactionDataProvider {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    override fun totalRecordCount(searchCriteria: HelpDeskSearchTransactionRequestDto): Mono<Int> =
+        when (searchCriteria) {
+            is SearchTransactionRequestPaymentTokenDto -> Mono.just(0)
+            is SearchTransactionRequestRptIdDto -> Mono.just(0)
+            is SearchTransactionRequestTransactionIdDto -> Mono.just(0)
```

Comment:
Why are these `0`?

Reply:
For PM we are only implementing search by email and user fiscal code.

Count return 0 in this case so that no findResult query is make. 

I implemented logic this way in order to abstract search mechanism for all params.

The main aim here is to make pagination against both databases search criteria agnostic. Single data provider instance returns 0 as total count for all that search criteria that has no scope for the current DB

Reply:
Map to error with https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/9/commits/aafebe6bdee6654562781935e34d66c75d9bb122